### ISSUE TITLE
Update package

### ISSUE THREAD

**Issue by lostella** [JuliaFirstOrder/StructuredOptimization.jl#38]
CI workflows fail since the package has long been unmaintained. In particular, I doubt it will support the latest `ProximalAlgorithms` versions.

**Comment by nantonel** [comment]
Yes, adding latest version of `ProximalAlgorithms` will take some time. I'm not sure how many people is actively using this package anymore.

**Comment by nantonel** [comment]
Wanted to elaborate my last comment here.  

Historically the goal of this package was to provide a user friendly interface to the other packages present in the JuliaFirstOrder ecosystem.

Now that ProximalAlgorithms supports an AD like Zygote I feel like StructOpt has lost its initial scope. If it is about using matrix free operators (AbstractOperators) these can still be used easily in ProximalAlotithms.

I think it's time we retire this package and invite new users to use ProximalAlgorithms only. I say this with some sadness as of course I'm quite emotionally attached to StructuredOptimization. I still remember when it was called RegLS!

So instead of updating this package I'd rather put the effort on moving the demos to ProximalAlgorithms. Let me know what you think @lostella

**Comment by lostella** [comment]
> So instead of updating this package I'd rather put the effort on moving the demos to ProximalAlgorithms

I definitely agree on this: all the demos would fit nicely in the documentation for ProximalAlgorithms. Sparse deconvolution, line spectral estimation, audio declipping, total variation denoising… Maybe even an example of optimal control problem

**Comment by hakkelt** [comment]
Hello @nantonel  and @lostella,
I really like the approach of this package, and I'm thinking about building an MRI reconstruction framework on top of it. I just created a PR that updates dependencies and makes all tests pass, but what do you think how difficult it would be to add support for multi-term algorithms like Davis-Yin or Chambolle-Pock? To have this, there should be some way to query the assumption of the selected solver, something like this:
```julia
julia> get_assumptions(ChambollePock)
(:(f + g + h ∘ L), Dict(:f => :is_smooth, :L => :is_linear, :h => :is_proximable, :g => :is_proximable))
```
Such a function could be added to ProximalAlgorithms for each solver, and then it feels feasible to implement a general enough solution to separate terms automatically.
